File: ELEC881_Tabnine/src/No Description Prompt/Python/q26-split-array-largest-sum.py
class Solution(object):
    def splitArray(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        total = sum(nums)
        if total % k != 0:
            return -1
        target = total // k
        dp = [float('inf')] * (target + 1)
        dp[0] = 0
        for num in nums:
            for j in range(target, num - 1, -1):
                dp[j] = min(dp[j], dp[j - num] + num)
        return dp[target] if dp[target] != float('inf') else -1




        

File: ELEC881_Tabnine/src/No Description Prompt/Java script (JS)/q26-split-array-largest-sum.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var splitArray = function(nums, k) {
    let sum = nums.reduce((acc, val) => acc + val, 0);
    let target = sum / k;

    let dp = Array.from({ length: nums.length + 1 }, () => Array.from({ length: k + 1 }, () => Infinity));
    dp[0][0] = 0;

    for (let i = 1; i <= nums.length; i++) {
        for (let j = 1; j <= k; j++) {
            for (let l = 0; l < i; l++) {
                dp[i][j] = Math.min(dp[i][j], dp[l][j - 1] + Math.max(0, nums[i - 1] - target * (j - 1) + target * (j - 2)));
            }
        }
    }

    return dp[nums.length][k];
};